project/src/main/webapp/script.js

```diff
@@ -21,28 +22,20 @@ function toggleNavBar() {
 
 /** Updates trend in carousel and respective dot based on next/previous buttons. */
 function switchTrend(val) {
```

Comment:
I could totally be misreading this but is there a difference between this function and updateCurrentSlide besides the input val? If not, could we simplify further by reusing the function?